### PR TITLE
Remove backoff in live test cleanup retry

### PR DESCRIPTION
**Pull request by benbp**
Addressing comment https://github.com/Azure/azure-sdk-tools/pull/3405#discussion_r891369015

### CODE REVIEW

File: eng/scripts/live-test-resource-cleanup.ps1 [hunk 1]
```diff
@@ -89,8 +89,6 @@ function Retry([scriptblock] $Action, [int] $Attempts = 5) {
             return $Action.Invoke()
         } catch {
             if ($attempt -lt $Attempts) {
-                $sleep *= 2
```
Comment (weshaggard) [inline eng/scripts/live-test-resource-cleanup.ps1:92]: Do you want to also remove this in the resource deployment scripts?

Comment (benbp) [reply eng/scripts/live-test-resource-cleanup.ps1:92]: Possibly. It looks like it's mostly wrapping `New-*` type calls, which we probably don't want backoff for anyway. I can do some testing.

Comment (benbp) [reply eng/scripts/live-test-resource-cleanup.ps1:92]: Also a test run for this PR took 30 minutes with a connection failure on china cloud, so I have to do some further digging.

**Review by JimSuplizio — COMMENTED** [review]
@benbp I do appreciate you shortening the wait time, though if 5 seconds isn't enough maybe going 10 seconds per wait isn't a terrible idea. 
There are 3 other files TestResources\New-TestResources.ps1, TestResources\Remove-TestResources.ps1, and TestResources\Update-TestResources.ps1 that all have this backoff timer,  should we also be fixing those?

**Comment by benbp** [comment]
@JimSuplizio @weshaggard I think 5 seconds will be fine for the error conditions that need to be retried. I'm going to defer backup timer changes on the test resources scripts because there is broader usage of retry in those scripts, it does not seem to be causing any glaring issues, and I'd bet there was a reason why the delay was added. @danieljurek I think this was you initially, do you have any context on why we needed backoff on retry in the resource provisioner script?

**Comment by weshaggard** [comment]
Another approach I sometimes take is to make the sleep equal to some constant times the iteration attempt. So maybe (1, 2, 3, 4, 5) seconds or (2, 4, 6, 8, 10) seconds. However either way I don't think it is going to make much of a difference.

**Review by weshaggard — APPROVED** [review]

**Comment by benbp** [comment]
> Another approach I sometimes take is to make the sleep equal to some constant times the iteration attempt. So maybe (1, 2, 3, 4, 5) seconds or (2, 4, 6, 8, 10) seconds. However either way I don't think it is going to make much of a difference.

Agreed if we change the sleep in test resources we should do something like this.

On a larger note, this speaks to our lack of common retry handlers, etc.

**Pull request merged**